vm/actor/src/builtin/miner/types.rs

```diff
@@ -9,105 +9,179 @@ use clock::ChainEpoch;
 use encoding::{serde_bytes, tuple::*};
 use fil_types::{PoStProof, RegisteredSealProof, SectorNumber};
 use num_bigint::bigint_ser;
-use vm::{DealID, TokenAmount};
+use vm::{DealID, Randomness, TokenAmount};
 
 pub type CronEvent = i64;
 pub const CRON_EVENT_WORKER_KEY_CHANGE: CronEvent = 1;
-pub const CRON_EVENT_PRE_COMMIT_EXPIRY: CronEvent = 2;
-pub const CRON_EVENT_PROVING_PERIOD: CronEvent = 3;
+pub const CRON_EVENT_PROVING_DEADLINE: CronEvent = 2;
+pub const CRON_EVENT_PROCESS_EARLY_TERMINATIONS: CronEvent = 3;
 
 /// Storage miner actor constructor params are defined here so the power actor can send them to the init actor
 /// to instantiate miners.
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct MinerConstructorParams {
     pub owner: Address,
     pub worker: Address,
+    pub control_addresses: Vec<Address>,
     pub seal_proof_type: RegisteredSealProof,
     #[serde(with = "serde_bytes")]
     pub peer_id: Vec<u8>,
     #[serde(with = "serde_bytes")]
     pub multi_address: Vec<u8>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct CronEventPayload {
     pub event_type: i64,
-    pub sectors: Option<BitField>,
 }
+
+#[derive(Debug)]
+pub struct PartitionKey {
+    pub deadline: u64,
+    pub partition: u64,
+}
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct GetControlAddressesReturn {
     pub owner: Address,
     pub worker: Address,
+    pub control_addresses: Vec<Address>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ChangeWorkerAddressParams {
     pub new_worker: Address,
+    pub new_control_addresses: Vec<Address>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ChangePeerIDParams {
     #[serde(with = "serde_bytes")]
     pub new_id: Vec<u8>,
 }
+
 #[derive(Serialize_tuple, Deserialize_tuple)]
 pub struct ChangeMultiaddrsParams {
     #[serde(with = "serde_bytes")]
     pub new_multi_addrs: Vec<u8>,
```

Comment:
This is wrong, but you'll come across it when you change the other usage of it, should be 2d vec